proxy/tests/transparency.rs

```diff
@@ -360,5 +368,242 @@ fn http1_get_doesnt_add_transfer_encoding() {
+    }
+}
+
+#[test]
+fn http1_bodyless_responses() {
+    let _ = env_logger::try_init();
+
+    let req_status_header = "x-test-status-requested";
+
+    let srv = server::http1()
+        .route_fn("/", move |req| {
+            let status = req.headers()
+                .get(req_status_header)
+                .map(|val| {
+                    val.to_str()
+                        .expect("req_status_header should be ascii")
+                        .parse::<u16>()
+                        .expect("req_status_header should be numbers")
+                })
+                .unwrap_or(200);
+
+            Response::builder()
+                .status(status)
+                .body(Default::default()) // default is hyper::Body::empty()
+                .unwrap()
+        })
+        .run();
+    let ctrl = controller::new().run();
+    let proxy = proxy::new()
+        .controller(ctrl)
+        .inbound(srv)
+        .run();
+    let client = client::http1(proxy.inbound, "transparency.test.svc.cluster.local");
+
+    // https://tools.ietf.org/html/rfc7230#section-3.3.3
+    // > response to a HEAD request, any 1xx, 204, or 304 cannot contain a body
+
+    //TODO: the proxy doesn't support CONNECT requests yet, but when we do,
+    //they should be tested here as well. As RFC7230 says, a 2xx response to
+    //a CONNECT request is not allowed to contain a body (but 4xx, 5xx can!).
+
+    let resp = client.request(
+        client
+            .request_builder("/")
+            .method("HEAD")
+    );
+
+    assert_eq!(resp.status(), StatusCode::OK);
+    assert!(!resp.headers().contains_key("content-length"));
+    assert!(!resp.headers().contains_key("transfer-encoding"));
+
+
+
+    let statuses = &[
+        //TODO: test some 1xx status codes.
+        //The current test server doesn't support sending 1xx responses
+        //easily. We could test this by making a new unit test with the
+        //server being a TCP server, and write the response manually.
+        StatusCode::NO_CONTENT, // 204
+        StatusCode::NOT_MODIFIED, // 304
+    ];
+
+    for &status in statuses {
+        let resp = client.request(
+            client
+                .request_builder("/")
+                .header(req_status_header, status.as_str())
+        );
+
+        assert_eq!(resp.status(), status);
+        assert!(!resp.headers().contains_key("content-length"), "content-length with status={:?}", status);
+        assert!(!resp.headers().contains_key("transfer-encoding"), "transfer-encoding with status={:?}", status);
+    }
+}
+
+#[test]
+fn http1_head_responses() {
+    let _ = env_logger::try_init();
+
+    let srv = server::http1()
+        .route_fn("/", move |req| {
+            assert_eq!(req.method(), "HEAD");
+            Response::builder()
+                .header("content-length", "55")
+                .body(Default::default()) // default is hyper::Body::empty()
```

Comment:
Hm, that comment is actually trying to say that `Default::default() == hyper::Body::empty()`. And as I write this, I cannot recall why I opted to use this with a comment instead of just `hyper::Body::empty()` directly.

Perhaps since `hyper::Body` isn't used in the http2 server cases, and so to allow the code to adapt?

Reply:
OK, I see I misread it. never mind my comments here.